src/Tools.Migration/Microsoft.Graph.Tools.Migration.psm1

```diff
@@ -0,0 +1,225 @@
+        $FileName = [System.IO.Path]::GetFileName($FilePath)
+        $UpdatedFile = ""
+        if (-not ([string]::IsNullOrEmpty($UpdatedFilePath))) {
+            $WriteToFile = $true
+            $UpdatedFile = Join-Path $UpdatedFilePath "Migration-$FileName"
+            if (-not (Test-Path $UpdatedFilePath)) { throw "Path does not exist: $UpdatedFilePath" }
+        }
+        $ScriptContent = Get-Content $FilePath
+        #Write-Host "Read-MgScript: $FilePath :$($scriptContent.Length)"
+        $Analysis = Invoke-MgScriptAnalyzer -ScriptContent $ScriptContent -GraphProfile $GraphProfile
+        $Result = @()
+        $i = 1
+        foreach ($item in $Original.GetEnumerator() | Sort Name) {
+            $LineNumber = $item.Key
+            $OriginalValue = $Original[$LineNumber].ToString()
+            $ProposedValue = $ProposedChanges[$LineNumber].ToString()
+            $Result += [pscustomobject]@{"Order" = $i;"Location" = $FileName+":"+$LineNumber; "Type"="Cmdlet"; "ApiVersion" = "Beta"; "Original" = $OriginalValue; "New" = $ProposedValue }
+            if ($WriteToFile) {
+                foreach ($Content in $ScriptContent) {
+
+                    if ($Content -eq $OriginalValue) {
+                        $ScriptContent = $ScriptContent | ForEach-Object { $_ -replace [regex]::Escape($OriginalValue), $ProposedValue }   
+                    }
+                    if ($Content.StartsWith("Select-MgProfile")) {
+                        $ScriptContent = $ScriptContent -replace $Content, $null
+                    }
+                     
+                }
+                $ScriptContent > $UpdatedFile
+            }
+            $i++
+        }
+        if ($Result.Count -gt 0) {
+            Write-Host -ForegroundColor Green "--------- Your script(s) contains commands that need to conform to the naming convention as per the 'New' column on the table below ---------"
+        }
+        Write-Host "counting stuff " $Result.Count
+        if($Result.Count -gt 0){
+        Write-Output $Result | Format-Table
+        }
+        if($WriteToFile){
+            Write-Host -ForegroundColor Blue "Your updated script is on this path: $UpdatedFile" 
+        }
+        $LineUpdate.Clear()
+        $Original.Clear()
+        $ProposedChanges.Clear()
+        $Result.Clear() 
+    }
+    #clear everything
+
+}
+
+function Invoke-MgScriptAnalyzer {
+    param (
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [ValidateNotNullorEmpty()]
+        [PSCustomObject] $ScriptContent,
+        [string] $GraphProfile
+    )
+
+    # TODO: Analyze script content for breaking change.
+    # TODO: Store changes in an object.
+    # TODO: Write object as output.
+    # TODO: Settle on object members.
+
+    # Proposed Output
+    # Order | Location (file name:line:tab) | Type (Cmdlet|CmdletParameter|ModuleName) | ApiVersion | Original | New
+ 
+    $i = 0
+    foreach ($_ in $ScriptContent) {
+        $LineContent.Add($_)
```

Comment:
Let's skip the line if it is commented out. i.e., starts with `#`. The logic should trim whitespaces.